File: backend/src/main/java/com/letrithien/ex1/entity/Product.java
package com.letrithien.ex1.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import org.hibernate.annotations.GenericGenerator;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "products")
@Data
public class Product {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    private UUID id;

    private String slug;

    @Column(name = "product_name")
    private String productName;

    private String sku;

    @Column(name = "sale_price")
    private BigDecimal salePrice;

    @Column(name = "compare_price")
    private BigDecimal comparePrice;

    @Column(name = "buying_price")
    private BigDecimal buyingPrice;

    private Integer quantity;

    @Column(name = "short_description")
    private String shortDescription;

    @Column(name = "product_description", columnDefinition = "TEXT")
    private String productDescription;

    @Column(name = "product_type")
    private String productType;

    @Column(columnDefinition = "TEXT")
    private String image;

    private Boolean published;

    @Column(name = "disable_out_of_stock")
    private Boolean disableOutOfStock;

    @Column(columnDefinition = "TEXT")
    private String note;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(name = "created_by")
    private UUID createdBy;

    @Column(name = "updated_by")
    private UUID updatedBy;
}

File: frontend/lib/models/product.dart
class Product {
  final String id;
  final String productName;
  final double salePrice;
  final int quantity;
  final String? shortDescription;
  final String? productType;
  final String? image;
  final double? comparePrice;
  final String? favoriteSize;

  Product({
    required this.id,
    required this.productName,
    required this.salePrice,
    required this.quantity,
    this.shortDescription,
    this.productType,
    this.image,
    this.comparePrice,
    this.favoriteSize,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id']?.toString() ?? '',
      productName: json['productName'] ?? 'Unknown Product',
      salePrice: (json['salePrice'] != null) ? (json['salePrice'] as num).toDouble() : 0.0,
      comparePrice: (json['comparePrice'] != null) ? (json['comparePrice'] as num).toDouble() : null,
      quantity: json['quantity'] ?? 0,
      shortDescription: json['shortDescription'],
      productType: json['productType'],
      image: json['image'],
      favoriteSize: json['favoriteSize'],
    );
  }
}
